fmriprep/workflows/anatomical.py

```diff
@@ -72,10 +74,131 @@ def t1w_preprocessing(name='t1w_preprocessing', settings=None):
+            name='BIDSInfo',
+            run_without_submitting=True)
+
+        autorecon1 = pe.Node(
+            freesurfer.ReconAll(
+                directive='autorecon1',
+                openmp=nthreads,
+                parallel=True),
+            name='Reconstruction')
+        autorecon1.interface._can_resume = False
+        autorecon1.interface.num_threads = nthreads
+
+        def inject_skullstripped(subjects_dir, subject_id, skullstripped):
+            import os
+            import nibabel as nib
+            from nilearn.image import resample_to_img, new_img_like
+            from nipype.utils.filemanip import copyfile
+            mridir = os.path.join(subjects_dir, subject_id, 'mri')
+            t1 = os.path.join(mridir, 'T1.mgz')
+            bm_auto = os.path.join(mridir, 'brainmask.auto.mgz')
+            bm = os.path.join(mridir, 'brainmask.mgz')
+
+            img = nib.load(t1)
+            mask = nib.load(skullstripped)
+            bmask = new_img_like(mask, mask.get_data() > 0)
+            resampled_mask = resample_to_img(bmask, img, 'nearest')
+            masked_image = new_img_like(img, img.get_data() * resampled_mask.get_data())
+            masked_image.to_filename(bm_auto)
+
+            copyfile(bm_auto, bm, copy=True, use_hardlink=True)
+            return subjects_dir, subject_id
+
+        injector = pe.Node(
+            niu.Function(
+                function=inject_skullstripped,
+                input_names=['subjects_dir', 'subject_id', 'skullstripped'],
+                output_names=['subjects_dir', 'subject_id']),
+            name='InjectSkullstrip')
+
+        reconall = pe.Node(
+            ReconAllRPT(
+                openmp=nthreads,
+                parallel=True,
+                out_report='reconall.svg',
+                generate_report=True),
+            name='Reconstruction2')
+        reconall.interface.num_threads = nthreads
+
+        recon_report = pe.Node(
+            DerivativesDataSink(base_directory=settings['output_dir'],
+                                suffix='reconall', out_path_base='reports'),
+            name='ReconAll_Report'
+        )
+
+        def joiner(dirname, basename):
+            import os
+            return os.path.join(dirname, basename)
+
+        subject_dir = pe.Node(
+            niu.Function(
+                function=joiner,
+                input_names=['dirname', 'basename'],
+                output_names=['path']),
+            name='SubjectDir',
+            run_without_submitting=True)
+
+        subject_out = pe.Node(
+            nio.DataSink(
+                base_directory=settings['output_dir'],
+                container='derivatives'),
+            name='FSSubjOut',
+            run_without_submitting=True)
+        # Copy fsaverage subject into derivatives/freesurfer
+        setattr(subject_out.inputs, 'freesurfer.@fsaverage',
+                '/opt/freesurfer/subjects/fsaverage')
```

Comment:
Working on `bbregister`/`mri_vol2surf`, I'm realizing we don't want this inside `t1w_preprocessing` -- or really anything below `workflow_enumerator` -- as then it would be re-run for every subject.

Does it make sense to make something like:

```Python
class BIDSFreeSurferDir(BaseInterface):
    inputs = (freesurfer_home, derivatives_dir)
    outputs = subjects_dir

    def _run_interface(self, runtime):
         # Create subjects directory
         # Copy fsaverage
```

And then pass `subjects_dir` through to workflows that need access to it? We would then be `recon-all`'ing directly in `derivatives/freesurfer/sub-{subj_id}/`, which seems okay to me.

WDYT, @chrisfilo? This can wait, but I think we'll have to deal with it in the next PR (where bbregister will need to know about the common subjects directory to do the registration) if we don't, now.